avocado/plugins/assets.py

```diff
@@ -279,6 +279,31 @@ def pre_tests(self, job):
             fetch_assets(*candidate, logger)
 
 
+class AssetPreTest(PreTest):
+    """Implements the asset pre tests plugin.
+
+    It will create pre-test tasks for fetching assets based on the
+     `:avocado: dependency=` definition inside the test’s docstring.
+    """
+    name = "asset requirement"
+    description = "Fetch of files using the Avocado Assets utility."
+
+    def pre_test_runnables(self, test_runnable):
```

Comment:
I like the use of the plugin architecture, but I don't like the code duplication, as this is essentially the same as the package plugin with a different type.

I can think of one of two possibilities:

1) Have a single plugin that simply handles all known dependencies types (packages, assets, etc)
2) Have a common method (or a `MixIn` class) that consolidates this code into a single method

Reply:
Sure, this needs refactoring. I would vote for second option, because then we will have more options how to manage dependencies. For example, the order of execution.